includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -514,6 +522,128 @@ public static function show_message_after_logout() {
+			$customer_id              = \get_current_user_id();
+			$memberships_info         = wc_memberships_get_user_active_memberships( $customer_id );
+			$memberships_without_subs = [];
+
+			// Create an array of active memberships without active subscriptions.
+			if ( function_exists( 'wc_memberships_has_subscription_product_granted_access' ) ) {
+				foreach ( $memberships_info as $membership ) {
+					if ( ! wc_memberships_has_subscription_product_granted_access( $membership ) ) {
+						$memberships_without_subs[] = $membership;
+					}
+				}
+			}
+
+			return $memberships_without_subs;
+		}
+	}
+
+	/**
+	 * Optionally append a table of active memberships without subscriptions on the My Account Subscriptions tab.
+	 */
+	public static function append_membership_table() {
+		// If this option is not enabled, stop.
+		if ( 'no' === get_option( 'wc_memberships_show_unsubbed_memberships', 'no' ) ) {
+			return;
+		}
+
+		$memberships_without_subs = self::get_memberships_without_subs();
+
+		// If there are active memberships without subscriptions, present them in a table.
+		if ( $memberships_without_subs ) {
+			echo '<div class="woocommerce-memberships-without-subs">';
+			echo '<h2>' . esc_html__( 'Active Memberships', 'newspack-plugin' ) . '</h2>';
+			echo '<p>' . esc_html__( 'These memberships are active, but don\'t have an associated subscription. They will need to be manually renewed when they expire.', 'newspack-plugin' ) . '</p>';
+			wc_get_template(
+				'myaccount/my-memberships.php',
+				array(
+					'customer_memberships' => $memberships_without_subs,
+					'user_id'              => get_current_user_id(),
+				)
+			);
+			echo '</div>';
+		}
+	}
+
+	/**
+	 * Returns whether or not to redirect the Subscriptions link to a single subscription, or to the main Subscriptions screen.
+	 *
+	 * @return bool
+	 */
+	public static function redirect_to_single_subscription() {
+		// If this option is not enabled, stop.
+		if ( 'no' === get_option( 'wc_memberships_show_unsubbed_memberships', 'no' ) ) {
+			return true;
+		}
+
+		$memberships_without_subs = self::get_memberships_without_subs();
+
+		// If there are memberships without subs, we want to remove the redirect and go to Subscriptions; otherwise, return true.
+		if ( $memberships_without_subs ) {
+			return false;
+		} else {
+			return true;
+		}
+	}
+
+	/**
+	 * Hides 'Cancel' button on main Memberships table to tidy it up.
+	 *
+	 * @param array $actions WooCommerce Memberships available actions.
+	 * @return array
+	 */
+	public static function hide_cancel_button_from_memberships_table( $actions ) {
+		unset( $actions['cancel'] );
+		return $actions;
+	}
+
+	/**
+	 * Removes the 'Next Bill On' column in the main Memberships table to tidy it up.
+	 *
+	 * @param array $columns WooCommerce Memberships table columns.
+	 * @return array
+	 */
+	public static function remove_next_bill_on( $columns ) {
+		unset( $columns['membership-next-bill-on'] );
```

Comment:
When reading a property of an array, it's always a good idea to check first whether the property exists in the array before reading it, otherwise it can and will throw a fatal error if the property doesn't exist. That's why you'll see this pattern everywhere in our code:

```suggestion
		if ( ! empty( $columns['membership-next-bill-on'] ) ) {
			unset( $columns['membership-next-bill-on'] );
		}	
```

Reply:
Thanks for catching this!